cleanlab/internal/validation.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (C) 2017-2022  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+
+"""
+Checks to ensure valid inputs for various methods.
+"""
+
+import warnings
+import numpy as np
+import pandas as pd
+from sklearn.utils import check_X_y
+
+
+def assert_valid_inputs(X, y, pred_probs=None):
+    """Checks that X, labels, and pred_probs are correctly formatted"""
+    if not isinstance(y, (list, np.ndarray, np.generic, pd.Series, pd.DataFrame)):
+        raise TypeError("labels should be a numpy array or pandas Series.")
+    y = labels_to_array(y)
+    assert_valid_class_labels(y)
+    allow_empty_X = False if pred_probs is None else True
+    if not allow_empty_X:
+        assert_nonempty_input(X)
+        try:
+            num_examples = len(X)
+            len_supported = True
+        except:
+            len_supported = False
+        if not len_supported:
+            try:
+                num_examples = X.shape[0]
+                shape_supported = True
+            except:
+                shape_supported = False
+        if (not len_supported) and (not shape_supported):
+            raise TypeError("Data features X must support either: len(X) or X.shape[0]")
+
+        if num_examples != len(y):
+            raise ValueError(
+                f"X and labels must be same length, but X is length {num_examples} and labels is length {len(y)}."
+            )
+
+        if num_examples < 2:
```

Comment:
is this even useful? shouldn't we check instead for at least two classes represented?

Reply:
i.e. `get_num_classes(labels) > 1`

Reply:
Good point, replaced it with this check inside `assert_valid_class_labels()`